### PR TITLE
OpenEXR 3.1.5 + rebuilds

### PR DESCRIPTION
**Pull request by cthbleachbit**
Topic Description
-----------------

This PR updates openexr to 3.1.4.

Package(s) Affected
-------------------

**Stage -1: missing packages on non amd64**

* [ ] exiv2 - depended by gegl-0.4 on riscv - no rebuilds necessary on other architectures, doesn't seem to build though
* [ ] spirv-tools - lag behind on arm64

**Stage 0: glslang used by gstreamer, upstreams switching URLs**

* [x] fftw - install `FFTW3LibraryDepends.cmake`
* [x] glslang
* [x] shaderc

**Stage 1: OpenEXR / OpenCV**

* [x] imath <- split from openexr, now independently versioned
* [x] openexr
* [x] opencv - no reverse dependency broken yet

**Stage 2: Rebuilds**

* [x] tdelibs
* [x] gegl-0.4
* [x] imagemagick
* [x] openimageio
* [x] mlt
* [x] mlt-6
* [x] waifu2x-converter-cpp
* [x] tdegraphics
* [x] vigra
* [x] opensubdiv
* [x] blender
* [ ] cinepaint (skipped - no idea how to build)
* [x] gst-plugins-bad-1-0
* [x] gimp
* [x] gmic - qt frontend and gimp plugins split
* [x] gmic-frontends - qt frontend and plugins
* [x] darktable
* ~~howdy - no rebuild required - no library linkage~~
* [x] hugin
* [x] imagemagick+7
* [x] kimageformats
* [x] kio-extras
* [x] koffice-trinity
* [x] krita
* [x] openscenegraph (updated - so ver broken)
* [x] povray
* [x] synfig
* [x] digikam
* [x] tdebase
* [x] [frei0r-plugins](https://packages.aosc.io/revdep/frei0r-plugins)
* [x] simgear (rebuild for openscenegraph)
* [x] flightgear (rebuild for openscenegraph)
* [x] openmw (rebuild for openscenegraph)

**Stage 3: Break dependencies**
* [x] opencv - break dependencies

Build Order
-----------

**Stage 0-2: EXCLUDE LAST COMMIT from stage 3**

fftw glslang shaderc imath openexr opencv tdelibs gegl-0.4 imagemagick openimageio mlt mlt-6 waifu2x-converter-cpp tdegraphics vigra opensubdiv blender ~~cinepaint~~ gst-plugins-bad-1-0 gimp gmic gmic-frontends darktable ~~howdy~~ hugin imagemagick+7 kimageformats kio-extras koffice-trinity krita openscenegraph povray synfig digikam tdebase frei0r-plugins simgear flightgear openmw

**Stage 3**

opencv

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

File: extra-libs/opencolorio/autobuild/prepare [hunk 1]
```diff
@@ -1,8 +1,4 @@
 export CXXFLAGS="${CXXFLAGS} -Wno-error=deprecated-declarations -Wno-error=unused-function"
-sed -e "s/push(hidden)/push(default)/g" \
-    -i "$SRCDIR"/src/core/OCIOYaml.cpp
 
-sed -e "s/-Werror//g" \
-    -i src/core/CMakeLists.txt
-sed -e "s/-Werror//g" \
-    -i src/pyglue/CMakeLists.txt
+abinfo "FIXME: cmake really likes /lib64/cmake/expat-VER, but there's no /include"
+sudo rm -rv /usr/lib/cmake/expat-*
```
Comment (MingcongBai) [inline extra-libs/opencolorio/autobuild/prepare:4]: Wrong path?

File: extra-libs/openscenegraph/autobuild/prepare [hunk 2]
```diff
@@ -1 +1 @@
-export CXXFLAGS="${CXXFLAGS} -Wno-narrowing"
+export CXXFLAGS="${CXXFLAGS} -Wno-narrowing -lGL"
```
Comment (MingcongBai) [inline extra-libs/openscenegraph/autobuild/prepare:2]: Missing abinfo.

File: extra-libs/opensubdiv/autobuild/prepare [hunk 3]
```diff
@@ -0,0 +1 @@
+export LDFLAGS="-lGLX"
```
Comment (MingcongBai) [inline extra-libs/opensubdiv/autobuild/prepare]: Missing abinfo, also, you are overriding LDFLAGS. Don't.

File: extra-libs/fftw/autobuild/build [hunk 4]
```diff
@@ -2,15 +2,32 @@ cd "$SRCDIR"/fftw-${PKGVER/+/-}
 CONFIGURE="./configure F77=gfortran --prefix=/usr \
            --enable-shared --enable-threads \
            --enable-openmp"
+CMAKE_OPTIONS="
+	-Bcmake-gen
+	-DCMAKE_INSTALL_PREFIX=/usr
+	-DCMAKE_BUILD_TYPE=None
```
Comment (MingcongBai) [inline extra-libs/fftw/autobuild/build:13]: Why?

File: extra-libs/fftw/autobuild/build [hunk 5]
```diff
@@ -2,15 +2,32 @@ cd "$SRCDIR"/fftw-${PKGVER/+/-}
...
+	-DENABLE_OPENMP=ON
+	-DENABLE_THREADS=ON
+"
 
 if [[ "${CROSS:-$ARCH}" = "amd64" ]]; then
-    CONF="--enable-sse3"
+    abinfo "FIXME: sse requires double precision"
```
Comment (MingcongBai) [inline extra-libs/fftw/autobuild/build]: Also clarify that you are not enabling SSE as a result.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-libs/fftw/autobuild/build (hunk 4 above)
Comment (cthbleachbit) [reply extra-libs/fftw/autobuild/build:13]: Cmake is required to generate `FFTW3LibraryDepends.cmake`

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**